lib/openshift4-monitoring-operator-rules.libsonnet

```
@@ -0,0 +1,198 @@
+  ],
+};
+
+/**
+ * \brief create a minimal ManagedResource for managing operator PrometheusRules.
+ *
+ * \arg ns
+ *        The namespace to deploy the ManagedResource to.
+ * \returns
+ *        A ManagedResource CR for managing operator PrometheusRules.
+ */
+local managedResource(ns) = esp.managedResource(global_name, ns) {
+  spec: {
+    serviceAccountRef: { name: serviceAccount(ns).metadata.name },
+    context: [
+      {
+        name: 'op_rules',
+        resource: {
+          apiVersion: 'monitoring.coreos.com/v1',
+          kind: 'PrometheusRule',
+          labelSelector: {
+            // Only match rules that are not created by espejote: 'espejote.io/created-by'
+            // and are not marked to be ignored: 'espejote.io/ignore'
+            matchExpressions: [
+              {
+                key: 'espejote.io/created-by',
+                operator: 'DoesNotExist',
+              },
+              {
+                key: 'espejote.io/ignore',
+                operator: 'DoesNotExist',
+              },
+            ],
+          },
+        },
+      },
+    ],
+    triggers: [
+      {
+        name: 'op_rules',
+        watchContextResource: {
+          name: 'op_rules',
+        },
+      },
+      {
+        name: 'generated_rules',
+        watchResource: {
+          apiVersion: 'monitoring.coreos.com/v1',
+          kind: 'PrometheusRule',
+          labelSelector: {
+            // Only match rules that are created by espejote: 'espejote.io/created-by=openshift4-monitoring-rules'
+            matchExpressions: [
+              {
+                key: 'espejote.io/created-by',
+                operator: 'In',
+                values: [ 'openshift4-monitoring-rules' ],
+              },
+            ],
+          },
+        },
+      },
+      {
+        name: 'jslib_global',
+        watchResource: {
+          apiVersion: esp_gv,
+          kind: 'JsonnetLibrary',
+          name: global_name,
+          namespace: global_namespace,
+        },
+      },
+      {
+        name: 'jslib_component',
+        watchResource: {
+          apiVersion: esp_gv,
+          kind: 'JsonnetLibrary',
+          name: global_name,
+          namespace: ns,
+        },
+      },
+    ],
+    template: |||
+      local esp = import 'espejote.libsonnet';
+
+      local renderer = import 'lib/openshift4-monitoring-rules/renderer_v1.libsonnet';
+      local configGlobal = import 'lib/openshift4-monitoring-rules/config_v1.json';
+      local configComponent = import 'openshift4-monitoring-rules/config.json';
+
+      local opRules = esp.context().op_rules;
+
+      if std.member([ 'op_rules', 'generated_rules' ], esp.triggerName) then [
+        // if the trigger is 'op_rules' or 'generated_rules', render single op_rule
+        local trigger = esp.triggerData();
+        if trigger != null && std.get(trigger.metadata, 'deletionTimestamp') != null then
+          renderer.process(or, configGlobal, configComponent)
+      ] else [
```

I'd restructure this part as follows (note that the suggestion also includes mandatory fixes):

```suggestion
      local inDelete(obj) = std.get(obj.metadata, 'deletionTimestamp', '') != '';
      if std.member([ 'op_rules', 'generated_rules' ], esp.triggerName) then 
        // if the trigger is 'op_rules' or 'generated_rules', render single op_rule
        local trigger = esp.triggerData();
        local or = if esp.triggerName == 'op_rules' then
          trigger.resource
        else
          local cand = std.filter(function(r) r.metadata.name == trigger.resource.metadata.ownerReferences[0].name, opRules)
          if std.length(cand) > 0 then cand[0];
        if trigger != null && !inDelete(trigger) && or != null && !inDelete(or) then
        [
          renderer.process(or, configGlobal, configComponent)
        ]
       else [
```